cmd/fleet/actionDispatcher.go

```diff
@@ -0,0 +1,120 @@
+package fleet
+
+import (
+	"context"
+	"fleet/internal/pkg/action"
+	"fleet/internal/pkg/esutil"
+	"fleet/internal/pkg/seqno"
+	"sync"
+
+	"github.com/rs/zerolog/log"
+)
+
+type ActionX = action.ActionX
+
+type ActionSubX struct {
+	agentId string
+	seqNo   int64
+	ch      chan []ActionX
+}
+
+func (as ActionSubX) Ch() chan []ActionX {
+	return as.ch
+}
+
+type ActionDispatcher struct {
+	am *seqno.Monitor
+
+	mx   sync.RWMutex
+	subs map[string]ActionSubX
+}
+
+func NewActionDispatcher(am *seqno.Monitor) *ActionDispatcher {
+	return &ActionDispatcher{
+		am:   am,
+		subs: make(map[string]ActionSubX),
+	}
+}
+
+func (ad *ActionDispatcher) Run(ctx context.Context) (err error) {
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case hits := <-ad.am.Output():
+			ad.process(ctx, hits)
+		}
+	}
+}
+
+func (ad *ActionDispatcher) Subscribe(agentId string, seqNo int64) *ActionSubX {
+	cbCh := make(chan []ActionX, 1)
```

Comment:
Being this is a buffered channel of 1, if the Agent doesn't quickly handle the action will this block the sending go routine affecting other Agent's?

Reply:
yes, there is a slim chance that sending to this channel can block because the agent long poll didn't exit and unsubscribed from the action dispatcher before the new action came in. 

That's on my TODO item below.
```
// TODO: revisit actions rollout
``` 

was focusing on getting this work end-to-end first, but actions rollout details are not fully fleshed out yet

Reply:
the easiest for now send with select default (drop the actions) for now so even if the channel is full it will not block.
and if the channel is full we should not dispatch anything anyways cause the agent has response at this point.